code/datums/components/run_over_atom.dm

```diff
@@ -0,0 +1,45 @@
+/datum/component/run_over_atom
+	var/run_over_message = "drives over"
+	var/damage_amount
+
+/datum/component/run_over_atom/Initialize(_damage, _runover_message)
+	if(!ismovableatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	damage_amount = _damage ? _damage : rand(5, 15) //see multipliers below.
+
+	if(_runover_message)
+		run_over_message = _runover_message
+	RegisterSignal(parent, COMSIG_MOVABLE_CROSSED_OVER, .proc/run_over)
+
+/datum/component/run_over_atom/proc/run_over(atom/movable/AM, atom/movable/thing_ran_over)
+	if(thing_ran_over.get_run_over(AM, damage_amount, run_over_message))
+		AM.after_running_over(thing_ran_over)
+
+/atom/movable/proc/after_running_over(atom/movable/AM)
+	return
+
+/mob/living/simple_animal/bot/mulebot/after_running_over(atom/movable/AM)
+	bloodiness += 4
+
+/atom/movable/proc/get_run_over(atom/movable/AM, _damage_amount, run_over_message)
+	return
+
+/mob/living/carbon/get_run_over(atom/movable/AM, _damage_amount, run_over_message)
```

Comment:
shouldn't this live on the datum, it seems like you're almost getting seperation of concerns but then falling down at the last steps

Reply:
Have an example of what you mean by chance? Doing this to figure out how components works, really.

Reply:
as in the component should deliver the damage to the object this proc shouldn't exist at all, or be a proc on the datum that just takes the target mob.